scheduler/src/scheduling.test.ts

```diff
@@ -1,121 +1,113 @@
+    expect(scheduler.getAllJobs()).toHaveLength(1)
+    const job123 = scheduler.getJob(123)
     expect(job123).toBeDefined()
-    if (job123 !== undefined) {
-      expect(job123.datasourceConfig).toEqual(toBeAdded)
-    }
+    // @ts-expect-error
+    expect(job123.datasourceConfig).toEqual(toBeAdded)
+    expect(mockedTriggerDatasource).toHaveBeenCalledTimes(1)
   })
 
   test('should apply deletion event', async () => {
-    const toBeDeleted = generateConfig(true, new Date(Date.now() + 5000), 6000)
+    const toBeDeleted = generateConfig(true, new Date(Date.now() + 500), 1000)
     mockedGetAllDatasources.mockResolvedValue([toBeDeleted])
-    const deletionEvent = {
-      eventId: 323,
-      eventType: EventType.DATASOURCE_DELETE,
-      datasourceId: 123
-    }
-
-    await Scheduling.initializeJobsWithRetry()
-    expect(Scheduling.getAllJobs()).toHaveLength(1)
+    await scheduler.initializeJobsWithRetry(CONNECTION_RETRIES, CONNECTION_BACKOFF_IN_MS)
 
-    mockedGetEventsAfter.mockResolvedValue([deletionEvent])
-    mockedGetDatasource.mockResolvedValue(toBeDeleted)
-    await Scheduling.updateDatasources()
+    const deletionEvent: DatasourceConfigEvent = {
+      datasource: toBeDeleted
+    }
+    await scheduler.applyDeleteEvent(deletionEvent)
 
-    await sleep(1000) // Give the scheduler some time to apply the update
+    await sleep(1000)
 
-    expect(Scheduling.getAllJobs()).toHaveLength(0)
+    expect(scheduler.getAllJobs()).toHaveLength(0)
+    expect(mockedTriggerDatasource).toHaveBeenCalledTimes(0)
   })
 
   test('should apply update event', async () => {
     const toBeUpdated = generateConfig(true, new Date(Date.now() + 5000), 6000)
     mockedGetAllDatasources.mockResolvedValue([toBeUpdated])
-    const updateEvent = {
-      eventId: 324,
-      eventType: EventType.DATASOURCE_UPDATE,
-      datasourceId: 123
-    }
 
-    await Scheduling.initializeJobsWithRetry()
-    const allJobs = Scheduling.getAllJobs()
-    expect(allJobs).toHaveLength(1)
-    expect(allJobs[0].datasourceConfig).toEqual(toBeUpdated)
+    await scheduler.initializeJobsWithRetry(CONNECTION_RETRIES, CONNECTION_BACKOFF_IN_MS)
 
     const updated = generateConfig(false, new Date(Date.now() + 5000), 12000)
-    mockedGetEventsAfter.mockResolvedValue([updateEvent])
-    mockedGetDatasource.mockResolvedValue(updated)
 
-    await Scheduling.updateDatasources()
+    const updateEvent: DatasourceConfigEvent = {
+      datasource: updated
+    }
 
-    await sleep(1000) // Give the scheduler some time to apply the update
+    await scheduler.applyCreateOrUpdateEvent(updateEvent)
 
-    expect(Scheduling.getAllJobs()).toHaveLength(1)
-    const job123 = Scheduling.getJob(123)
+    expect(scheduler.getAllJobs()).toHaveLength(1)
+    const job123 = scheduler.getJob(123)
     expect(job123).toBeDefined()
-    if (job123 !== undefined) {
-      expect(job123.datasourceConfig).toEqual(updated)
-    }
+    // @ts-expect-error
```

Comment:
I looked again at this problem. It seams like Optional Chaining is the better solution instead of the suppression of the TypeScript warning.

```js
    expect(job123?.datasourceConfig).toEqual(toBeAdded)
    // if job123 or the attribute datasourceConfig is not defined it will return 'undefined'
```